torch/csrc/profiler/kineto_client_interface.cpp

```diff
@@ -18,16 +18,29 @@ class LibKinetoClient : public libkineto::ClientInterface {
     reportInputShapes_ = setupOpInputsCollection;
   }
 
+#ifdef USE_KINETO_UPDATED
```

Comment:
We should try to minimize the use of `#ifdef`, especially when it alters a function signature. In `kineto_shim.h` we convert these `#ifdef USE_KINETO` into a constexpr bool (https://github.com/pytorch/pytorch/blob/master/torch/csrc/profiler/kineto_shim.h#L32). I think you should do the same with `USE_KINETO_UPDATED` and then use a more C++ idiomatic way of guarding. Do we care about compile time vs. runtime check? If the latter is enough you can just do something like `TORCH_CHECK(!withStack || kUseKinetoUpdated);`. If you want compile time check you'll have to do something like using a template to make `withStack` constexpr.